src/utils.rs

```diff
@@ -19,3 +21,292 @@ impl<T> PyResultMethods<T> for PyResult<T> {
+    "ab",
+    "ad",
+    "nemo",
+    "fugit",
+    "officia",
+    "at",
+    "in",
+    "id",
+    "quos",
+    "reprehenderit",
+    "numquam",
+    "iste",
+    "fugiat",
+    "sit",
+    "inventore",
+    "beatae",
+    "repellendus",
+    "magnam",
+    "recusandae",
+    "quod",
+    "explicabo",
+    "doloremque",
+    "aperiam",
+    "consequatur",
+    "asperiores",
+    "commodi",
+    "option",
+    "dolor",
+    "labore",
+    "temporibus",
+    "repellat",
+    "veniam",
+    "architecto",
+    "est",
+    "esse",
+    "mollitia",
+    "nulla",
+    "a",
+    "similique",
+    "eos",
+    "alias",
+    "dolore",
+    "tenetur",
+    "deleniti",
+    "porro",
+    "facere",
+    "maxime",
+    "corrupti",
+];
+
+static COMMON_WORDS: [&str; 19] = [
+    "lorem",
+    "ipsum",
+    "dolor",
+    "sit",
+    "amet",
+    "consectetur",
+    "adipisicing",
+    "elit",
+    "sed",
+    "do",
+    "eiusmod",
+    "tempor",
+    "incididunt",
+    "ut",
+    "labore",
+    "et",
+    "dolore",
+    "magna",
+    "aliqua",
+];
+
+pub fn sentence() -> String {
+    let mut rng = rand::thread_rng();
+    let num_sections = rng.gen_range(1..=5);
+    let mut sections = Vec::new();
+
+    for _ in 0..num_sections {
+        let num_words = rng.gen_range(3..=12);
+        let selected_words: Vec<&str> = WORDS
+            .choose_multiple(&mut rng, num_words)
+            .cloned()
+            .collect();
+        sections.push(selected_words.join(" "));
+    }
+
+    let sentence = sections.join(", ");
+    let first_char = sentence
+        .chars()
+        .next()
+        .unwrap()
+        .to_uppercase()
+        .collect::<String>();
+    let rest = &sentence[1..];
+    let punctuation = if rng.gen_bool(0.5) { "?" } else { "." };
+
+    format!("{}{}{}", first_char, rest, punctuation)
+}
+
+pub fn paragraph() -> String {
+    let num_sentences = rand::thread_rng().gen_range(1..=4);
+    (0..num_sentences)
+        .map(|_| sentence())
+        .collect::<Vec<String>>()
+        .join(" ")
+}
+
+pub fn paragraphs(count: usize, common: bool) -> Vec<String> {
+    let mut paras = Vec::new();
+
+    for i in 0..count {
+        if common && i == 0 {
+            paras.push(COMMON_P.to_string());
+        } else {
+            paras.push(paragraph());
+        }
+    }
+
+    paras
+}
+
+pub fn words(count: usize, common: bool) -> String {
+    let mut rng = rand::thread_rng();
+
+    let mut word_list: Vec<&str> = if common {
+        COMMON_WORDS.to_vec()
+    } else {
+        Vec::new()
+    };
+
+    let mut remaining = count;
+
+    if remaining <= word_list.len() {
+        word_list.truncate(remaining);
+    } else {
+        remaining -= word_list.len();
+
+        while remaining > 0 {
+            let take = remaining.min(WORDS.len());
+            let sampled = WORDS.choose_multiple(&mut rng, take);
+            word_list.extend(sampled.cloned());
+            remaining -= take;
+        }
+    }
+
+    word_list.join(" ")
```

Comment:
I don't think we need to create a `Vec` from `COMMON_WORDS` if `count` is less than `COMMON_WORDS.len()`:

```suggestion
pub fn words(count: usize, common: bool) -> String {
    if common && count <= COMMON_WORDS.len() {
        return COMMON_WORDS[..count].join(" ");
    }

    let mut rng = rand::thread_rng();
    let mut word_list: Vec<&str> = if common {
        COMMON_WORDS.to_vec()
    } else {
        Vec::new()
    };

    let mut remaining = count - word_list.len();

    while remaining > 0 {
        let take = remaining.min(WORDS.len());
        let sampled = WORDS.choose_multiple(&mut rng, take);
        word_list.extend(sampled.cloned());
        remaining -= take;
    }

    word_list.join(" ")
```